### PR TITLE
Package/py python htmlgen

### PR DESCRIPTION
**Pull request by Sinan81**
New package, dependency for #9661

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-python-htmlgen/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    """
+
+    homepage = "https://github.com/srittau/python-htmlgen"
+    url      = "https://github.com/srittau/python-htmlgen/archive/v1.2.2.tar.gz"
+
+    version('1.2.2', sha256='9dc60e10511f0fd13014659514c6c333498c21779173deb585cd4964ea667770')
+    conflicts('python@3.0:3.3.99')
+    depends_on('py-setuptools', type='build')
+    # dependencies for tests
+    depends_on('py-typing', type='build')
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-python-htmlgen/package.py]: there is a `test` dependency

File: var/spack/repos/builtin/packages/py-python-htmlgen/package.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+from spack import *
+
+
+class PyPythonHtmlgen(PythonPackage):
+    """Library to generate HTML from classes.
+    """
+
+    homepage = "https://github.com/srittau/python-htmlgen"
+    url      = "https://github.com/srittau/python-htmlgen/archive/v1.2.2.tar.gz"
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-python-htmlgen/package.py]: I think we prefer pypi.io over github urls for python packages. But no strong feelings

File: var/spack/repos/builtin/packages/py-python-htmlgen/package.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    version('1.2.2', sha256='9dc60e10511f0fd13014659514c6c333498c21779173deb585cd4964ea667770')
+    conflicts('python@3.0:3.3.99')
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-python-htmlgen/package.py]: would you satisfy my OCD and add empty lines around this conflicts?

File: var/spack/repos/builtin/packages/py-python-htmlgen/package.py (hunk 2 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-python-htmlgen/package.py]: It doesn't seem like source files are available under pypi.io: https://pypi.org/project/htmlgen/#files So, I will keep the url if that is ok.

File: var/spack/repos/builtin/packages/py-python-htmlgen/package.py (hunk 3 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-python-htmlgen/package.py]:
```suggestion

    conflicts('python@3.0:3.3.99')

```

File: var/spack/repos/builtin/packages/py-python-htmlgen/package.py (hunk 1 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-python-htmlgen/package.py]:
```suggestion
    depends_on('py-typing', type='test')
```

**Pull request merged**